clevertap-core/src/main/java/com/clevertap/android/sdk/events/EventQueueManager.java

```diff
@@ -455,16 +458,18 @@ public Future<?> queueEvent(final Context context, final JSONObject event, final
             @Override
             @WorkerThread
             public Void call() {
-
+                String eventName = eventMediator.getEventName(event);
                 Location userLocation = cleverTapMetaData.getLocationFromUser();
 
+                updateLocalStore(eventName, eventType);
```

Comment:
I believe updating the local store needs to happen in `processEvent`, together with the evaluation. The storing of the event and evaluation needs to happen only if the event is not dropped. This is also in parity when the event is stored in the event database.
The above also correlates with how it is implemented in the iOS SDK.

Example changes:
- remove the update local store and evaluation (lines 464-484)
- add the above code into the `processEvent`:
```
public void processEvent(final Context context, final JSONObject event, final int eventType) {
        synchronized (ctLockManager.getEventLock()) {
            try {
                if (CoreMetaData.getActivityCount() == 0) {
                    CoreMetaData.setActivityCount(1);
                }
                String type;
                if (eventType == Constants.PAGE_EVENT) {
                    type = "page";
                } else if (eventType == Constants.PING_EVENT) {
                    type = "ping";
                    attachMeta(event, context);
                    if (event.has("bk")) {
                        cleverTapMetaData.setBgPing(true);
                        event.remove("bk");
                    }

                    //Add a flag to denote, PING event is for geofences
                    if (cleverTapMetaData.isLocationForGeofence()) {
                        event.put("gf", true);
                        cleverTapMetaData.setLocationForGeofence(false);
                        event.put("gfSDKVersion", cleverTapMetaData.getGeofenceSDKVersion());
                        cleverTapMetaData.setGeofenceSDKVersion(0);
                    }
                } else if (eventType == Constants.PROFILE_EVENT) {
                    type = "profile";
                } else if (eventType == Constants.DATA_EVENT) {
                    type = "data";
                } else {
                    type = "event";
                }

                // Complete the received event with the other params

                String currentActivityName = cleverTapMetaData.getScreenName();
                if (currentActivityName != null) {
                    event.put("n", currentActivityName);
                }

                int session = cleverTapMetaData.getCurrentSessionId();
                event.put("s", session);
                event.put("pg", CoreMetaData.getActivityCount());
                event.put("type", type);
                event.put("ep", getNow());
                event.put("f", cleverTapMetaData.isFirstSession());
                event.put("lsl", cleverTapMetaData.getLastSessionLength());
                attachPackageNameIfRequired(context, event);

                // Report any pending validation error
                ValidationResult vr = validationResultStack.popValidationResult();
                if (vr != null) {
                    event.put(Constants.ERROR_KEY, getErrorObject(vr));
                }
                localDataStore.setDataSyncFlag(event);
                baseDatabaseManager.queueEventToDB(context, event, eventType);

                String eventName = eventMediator.getEventName(event);
                Location userLocation = cleverTapMetaData.getLocationFromUser();

                updateLocalStore(eventName, eventType);

                if (eventMediator.isChargedEvent(event)) {
                    controllerManager.getInAppController()
                            .onQueueChargedEvent(eventMediator.getChargedEventDetails(event),
                                    eventMediator.getChargedEventItemDetails(event), userLocation);
                } else if (!NetworkManager.isNetworkOnline(context) && eventMediator.isEvent(event)) {
                    // in case device is offline just evaluate all events
                    controllerManager.getInAppController().onQueueEvent(eventName,
                            eventMediator.getEventProperties(event), userLocation);
                } else if (eventType == Constants.PROFILE_EVENT) {
                    // in case profile event, evaluate for user attribute changes
                    Map<String, Map<String, Object>> userAttributeChangedProperties
                            = eventMediator.computeUserAttributeChangeProperties(event);
                    controllerManager.getInAppController()
                            .onQueueProfileEvent(userAttributeChangedProperties, userLocation);
                } else if (!eventMediator.isAppLaunchedEvent(event) && eventMediator.isEvent(event)) {
                    // in case device is online only evaluate non-appLaunched events
                    controllerManager.getInAppController().onQueueEvent(eventName,
                            eventMediator.getEventProperties(event), userLocation);
                }

                scheduleQueueFlush(context);

            } catch (Throwable e) {
                config.getLogger().verbose(config.getAccountId(), "Failed to queue event: " + event.toString(), e);
            }
        }
    }
```

Fix the `test_queueEvent_for_profileEvent_will_process_further_and_add_to_queue_when_event_should_not_be_dropped` test, remove:
```
`doNothing().`when`(eventQueueManager).addToQueue(application, json, Constants.PROFILE_EVENT)`.
``` 
and check if any additional cleanup needs to be done.

Reply:
Here the understanding is that we capture this as per local state. Keep this same in android and IOS. I believe that even if app records 5 event sent locally there might be failure in network and no events are recorded [they will be sent in future retries]

is the code snippet which you pasted synonymous to IOS impl? then we can take care of changes.

Reply:
resolved by @vasct